### PR TITLE
HIC and PIT rewrite

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/grda_warehouse/hud/funder.rb [hunk 1]
```diff
@@ -87,6 +87,21 @@ def self.related_item_keys
...
     end
 
+    def for_export
+      # This should never happen, but does
+      self.ProjectID = if self.ProjectID.blank?
+        'Unknown'
+      else
+        project&.id
+      end
+      self.GrantID ||= 'Unknown'
```
Comment (awisema) [inline app/models/grda_warehouse/hud/funder.rb:93]: `''` is OK here?

File: app/models/grda_warehouse/hud/funder.rb [hunk 2]
```diff
@@ -87,6 +87,21 @@ def self.related_item_keys
       [:ProjectID]
     end
 
+    def for_export
+      # This should never happen, but does
+      self.ProjectID = if self.ProjectID.blank?
+        'Unknown'
+      else
+        project&.id
+      end
```
Comment (awisema) [inline app/models/grda_warehouse/hud/funder.rb]:
```suggestion
      self.ProjectID =  project&.id || 'Unknown'
```

File: app/models/grda_warehouse/hud/inventory.rb [hunk 3]
```diff
@@ -76,6 +87,26 @@ class Inventory < Base
       where(HouseholdType: HOUSEHOLD_TYPES[:child_only])
     end
 
+    def for_export
+      # This should never happen, but does
+      self.ProjectID = if self.ProjectID.blank?
+        'Unknown'
+      else
+        project&.id
+      end
```
Comment (awisema) [inline app/models/grda_warehouse/hud/inventory.rb]:
```suggestion
      self.ProjectID = project&.id || 'Unknown'
```

File: app/models/grda_warehouse/hud/project.rb [hunk 4]
```diff
@@ -658,6 +658,43 @@ def self.export_providers(coc_codes)
       end
     end
 
+    def for_export
+      # This should never happen, but does
+      self.OrganizationID = if self.OrganizationID.blank?
+        'Unknown'
+      else
+        organization&.id
+      end
```
Comment (awisema) [inline app/models/grda_warehouse/hud/project.rb]:
```suggestion
      # This should never happen, but does
      self.OrganizationID = organization&.id || 'Unknown'
```

File: drivers/hud_hic/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+## HudHic README
+
+This README file should be used to explain the functionality of the driver.
```
Comment (awisema) [inline drivers/hud_hic/README.md]:
```suggestion
HUD Report to generate a zip file of CSVs to unpack and upload for the HUD HIC. As of the 2022 HIC, there are no programing specs.
```

File: app/views/hud_reports/_table.haml [hunk 6]
```diff
@@ -1,4 +1,5 @@
 :ruby
+  link = true unless defined?(link)
```
Comment (awisema) [inline app/views/hud_reports/_table.haml]: Rename to `show_link`?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/grda_warehouse/hud/funder.rb (hunk 1 above)
Comment (eanders) [reply app/models/grda_warehouse/hud/funder.rb:93]: Null False according to the Spec.

**Pull request merged**